### PR TITLE
CI: Update rtdv3 verify and merge workflows

### PR DESCRIPTION
**Pull request by jwagantall**
Allow the rtdv3 workflows to be used as required or be used at the repository level.

### CODE REVIEW

File: .github/workflows/gerrit-compose-required-rtdv3-merge.yaml [hunk 1]
```diff
@@ -41,7 +41,15 @@ on:
         description: "Gerrit refspec of change"
         required: true
         type: string
+      TARGET_REPO:
+        # yamllint disable-line rule:line-length
+        description: "The target GitHub repository needing the required workflow"
+        required: true
```
Comment (tykeal) [inline .github/workflows/gerrit-compose-required-rtdv3-merge.yaml:47]:
```
required: false
default: ${{ github.repository }}
```

File: .github/workflows/gerrit-compose-required-rtdv3-merge.yaml [hunk 2]
```diff
@@ -41,7 +41,15 @@ on:
...
+        type: string
     secrets:
+      GERRIT_SSH_REQUIRED_PRIVKEY:
```
Comment (tykeal) [inline .github/workflows/gerrit-compose-required-rtdv3-merge.yaml:50]: Not needed

File: .github/workflows/gerrit-compose-required-rtdv3-verify.yaml [hunk 3]
```diff
@@ -41,7 +41,15 @@ on:
         description: "Gerrit refspec of change"
         required: true
         type: string
+      TARGET_REPO:
+        # yamllint disable-line rule:line-length
+        description: "The target GitHub repository needing the required workflow"
+        required: true
```
Comment (tykeal) [inline .github/workflows/gerrit-compose-required-rtdv3-verify.yaml:47]:
```
required: false
default: ${{ github.repository }}
```

File: .github/workflows/gerrit-compose-required-rtdv3-verify.yaml [hunk 4]
```diff
@@ -41,7 +41,15 @@ on:
...
+        type: string
     secrets:
+      GERRIT_SSH_REQUIRED_PRIVKEY:
```
Comment (tykeal) [inline .github/workflows/gerrit-compose-required-rtdv3-verify.yaml:50]: Not needed

**Review by tykeal — CHANGES_REQUESTED** [review]
Doing what you're doing isn't going  to be conducive to projects that do not want to use this as a required workflow across their organization (for instance, ODL doesn't not dictate that this job runs across their org, but some projects may use it)

Also, while reviewing this change I realized that the `rtd-verify` job is doing a verify, which always uses a Gerrit Change, but the change is going to be merged at this point and it should be validating on the branch head... not certain how to reconcile that properly.

Given the changes that I'm requesting, you also need to make sure that the checkout action in use also adds in the following with:

```
repository: ${{ inputs.TARGET_REPO }}
```

File: .github/workflows/gerrit-compose-required-rtdv3-merge.yaml (hunk 2 above)
Comment (jwagantall) [reply .github/workflows/gerrit-compose-required-rtdv3-merge.yaml:50]: done

File: .github/workflows/gerrit-compose-required-rtdv3-verify.yaml (hunk 3 above)
Comment (jwagantall) [reply .github/workflows/gerrit-compose-required-rtdv3-verify.yaml:47]: done

File: .github/workflows/gerrit-compose-required-rtdv3-merge.yaml (hunk 1 above)
Comment (jwagantall) [reply .github/workflows/gerrit-compose-required-rtdv3-merge.yaml:47]: done

File: .github/workflows/gerrit-compose-required-rtdv3-verify.yaml (hunk 4 above)
Comment (jwagantall) [reply .github/workflows/gerrit-compose-required-rtdv3-verify.yaml:50]: done

**Review by tykeal — CHANGES_REQUESTED** [review]
Should probably change the title of all of this. This change is making it possible to be used as either required or used at the repository level

**Review by tykeal — APPROVED** [review]

**Pull request merged**